### PR TITLE
Allow indicating a format version that should be used to write/read from (De)serializer and use it for plans

### PR DESCRIPTION
**Pull request by bleskes**
#4420 introduced the ability to serialize and deserialize query plans. One of the intended use cases is to send query plans over to another DuckDB process and back (see test case in that PR). This is likely to be used in cases where these two processes don't use the same DuckDB version. As such, we want to have a mechanism for them to negotiate a serialization version and use that when reading and writing to each other. 

This PR adds such a mechanism and includes a test case with example usage.

### CODE REVIEW

File: src/include/duckdb/planner/logical_operator.hpp [hunk 1]
```diff
@@ -24,6 +24,10 @@ namespace duckdb {
 class FieldWriter;
 class FieldReader;
 
+//! The current version of the plan serialization format. Exposed via by @Serializer & @Desializer
```
Comment (Mause) [inline src/include/duckdb/planner/logical_operator.hpp:27]:
```suggestion
//! The current version of the plan serialization format. Exposed via by @Serializer & @Deserializer
```

File: test/api/test_plan_serialization.cpp [hunk 2]
```diff
@@ -44,11 +44,13 @@ static void test_helper(string sql, vector<string> fixtures = vector<string>())
 		//		printf("[%d] Optimized plan\n", i);
 
 		BufferedSerializer serializer;
+		serializer.SetVersion(PLAN_SERIALIZATION_VERSION);
 		plan->Serialize(serializer);
 		//		printf("[%d] Serialized plan\n", i);
```
Comment (Mause) [inline test/api/test_plan_serialization.cpp:49]:
```suggestion
```

**Review by Mause — CHANGES_REQUESTED** [review]

File: src/include/duckdb/planner/logical_operator.hpp (hunk 1 above)
Comment (bleskes) [reply src/include/duckdb/planner/logical_operator.hpp]: Thanks @Mause !

File: test/api/test_plan_serialization.cpp (hunk 2 above)
Comment (bleskes) [reply test/api/test_plan_serialization.cpp:49]: Thanks! I rather not change code that's not part of this PR so prefer to leave this line in place.

**Comment by bleskes** [comment]
It took a while to stabilize, but I think this PR is ready. The two failing tests seem unrelated, perhaps they can be re-run?

**Review by hannes — APPROVED** [review]

**Comment by hannes** [comment]
So we should add a test that serializes a ton of query plans to a file that we check in, and then check that those can still be cleanly deserialised, similar to what we do for the storage version. That way we are less likely to accidentally change the serialisation without incrementing the version

**Comment by bleskes** [comment]
@hannes sure thing. Shall we do it in a follow up PR?

**Comment by hannes** [comment]
Thanks!

**Pull request merged**